### DISCUSSION TITLE
Different types when I am exporting from the API & when using on the client side.

### DISCUSSION THREAD

**Discussion by HarshitKmr10** [honojs/hono#2900] [Q&A]
Hi everyone,

I'm building an application using turborepo with the following structure:

- apps
  - api (Hono API)
  - web (Next.js)
- packages
  - db
  - data-access

I'm keen on utilizing Hono's RPC functionality to export API route types from the api app and integrate them into my Next.js application in the web app. The issue is, the types are different when I am exporting from the API and when I am using on the client side. Here are the relevant screenshots for assistance:
![WhatsApp Image 2024-05-21 at 20 00 05](https://github.com/honojs/hono/assets/97459403/e5e08021-91f9-4cfc-a402-d4073e4d7922)
![WhatsApp Image 2024-05-21 at 20 00 14](https://github.com/honojs/hono/assets/97459403/e8ec995c-a595-4995-b419-29fae84203d3)

**Comment by EdamAme-x** [comment]
please show me minimal project?

**Comment by Chaitanya134** [comment]
Hi, I faced a similar issue. I think it arises when using `paths` in the tsconfig.json file in both the apps.

I fixed it by updating the paths in the apps/web/tsconfig.json file to include the path to the api folder's src.

```
// apps/api/tsconfig.json
{
  // Other configurations...
  "compilerOptions": {
    "paths": {
      "@/*": ["./src/*"]
    }
  }
}

// apps/web/tsconfig.json
{
  // Other configurations...
  "compilerOptions": {
    "paths": {
      "@/*": ["./src/*", "../api/src/*"]
    }
  }
}
```

**Reply by fstodulski** [reply]
You saved my day.
